JS/Lesson34/goksel/recipe-app/src/modules/user/RegistrationForm/index.tsx

```diff
@@ -0,0 +1,187 @@
+import { useForm } from "react-hook-form";
+import { Link, unstable_HistoryRouter } from 'react-router-dom';
+import Avatar from '@mui/material/Avatar';
+import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
+import { Box, Container, Grid, TextField, Typography, FormControl, InputLabel, OutlinedInput, InputAdornment, IconButton, FormControlLabel, Button, Snackbar } from "@mui/material";
+import Visibility from '@mui/icons-material/Visibility';
+import VisibilityOff from '@mui/icons-material/VisibilityOff';
+import * as React from 'react';
+import Checkbox from '@mui/material/Checkbox';
+
+
+interface RegistrationFormProps {
+	open: boolean;
+	setOpen: any;
+	firstName: string;
+	lastName: string;
+	email:string;
+	password:string;
+}
+
+export const RegistrationForm = () => {
+
+	const [open, setOpen] =React.useState(false);
+
+	const {
+		register,
+		handleSubmit,
+		formState: { errors },
+		reset
+	} = useForm<RegistrationFormProps>();
+
+
+	const onSubmit = (data:RegistrationFormProps) => {	
+		console.log(data);
+		reset();
+		setOpen(true);
+	}
+
+	const [showPassword, setShowPassword] =React.useState(false);
+
+	const handleClickShowPassword = () => setShowPassword(!showPassword);
+
+	return (
+
+		<Container component="main" maxWidth="xs">
+			<Box sx={{ display: "flex", flexDirection: "column", alignItems: "center" }}>
+				<Avatar sx={{ mt: 8, bgcolor: "primary.main" }}>
+					<LockOutlinedIcon />
+				</Avatar>
+				<Typography component="h1" variant='h5'>Sign Up</Typography>
+
+				<Box component="form" noValidate sx={{ mt: 3 }} onSubmit={handleSubmit(onSubmit)}>
+
+					<Grid container spacing={2}>
+						<Grid item xs={12} sm={6}>
+							<TextField
+								required
+								fullWidth
+								//@ts-ignore
+								name="firstName"
```

Comment:
Since you use 'react-hook-form' and in particular `register` function for the Text fields, adding `name` attribute causes an error. In fact, the code tries to specify name attribute twice for the same field: 
- directly by applying `name` attribute
- indirectly by using `register` function

Reply:
So remove both `//@ts-ignore` and `name="firstName"`, do the same for other text fields.

Reply:
Fixed

Reply:
The name is still here, remove both ts-ignore and the name attribute

Reply:
Sorry ı've forget the second one.